### ISSUE TITLE
IndexOutOfBoundsException

### ISSUE THREAD

**Issue by m-harisqurashi** [mapbox/mapbox-navigation-android#639]
When using `NavigationView ` and start navigation, app crashes a lot. Check below for crash log.

```
java.lang.IndexOutOfBoundsException: Index: 0, Size: 0
	at java.util.ArrayList.get(ArrayList.java:437)
	at com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationNotification.updateInstructionText(MapboxNavigationNotification.java:149)
	at com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationNotification.updateNotificationViews(MapboxNavigationNotification.java:132)
	at com.mapbox.services.android.navigation.v5.navigation.MapboxNavigationNotification.updateNotification(MapboxNavigationNotification.java:68)
	at com.mapbox.services.android.navigation.v5.navigation.NavigationService.onNewRouteProgress(NavigationService.java:115)
	at com.mapbox.services.android.navigation.v5.navigation.NavigationEngine$1.run(NavigationEngine.java:83)
	at android.os.Handler.handleCallback(Handler.java:790)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:164)
	at android.app.ActivityThread.main(ActivityThread.java:6494)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
```

**Comment by m-harisqurashi** [comment]
The crash issue happens when it is opened a second time and it is happening in some other scenarios as well.

**Comment by danesfeder** [comment]
@m-harisqurashi Thanks for bringing this to us.  https://github.com/mapbox/mapbox-navigation-android/pull/637 landed yesterday and should address this issue.  It's currently available with the `0.8.1-SNAPSHOT` and will be available in our upcoming stable release.
